.github/workflows/reusable-setup-cache.yml

```diff
@@ -50,6 +50,8 @@ jobs:
         with:
           # only install binary if cypress cache missed
           skip-cypress-binary: ${{ steps.restore-cypress-cache.outputs.cache-hit }}
+      - name: Validate licenses
+        run: yarn license
```

Comment:
Can we instead run this on 

`test-fork-prs.yml` and `test-internal-prs.yml`? That is where miscallenous tests go.

Reply:
Maybe run this on `reusable-unit` is a better choice? That way we do not need to repeat the same logic in 2 files and block invalid license earlier

Reply:
That is fair, but I think `reusable-unit` should only consist of unit test.

Note that we can still have license tests run before e2e in `test-fork-prs.yml` and `test-internal-prs.yml`, so we don't have to worry about them not failing early.

You are right however that we'll need to duplicate the logic, and we've already done this for `codeql` and `dependency-review`.  It's not ideal, but I think it's a tradeoff that we can make for the purpose of keeping fork/internal workflows separated. In the future, maybe we can create another reusable workflow that collects those miscallenous PR checks? (codeql, dependency-review, and license-checker)

Reply:
Sounds good. It makes sense to put those miscellaneous checks together in the future.

Reply:
Agreed, I'll add a task for that. Thanks!